services/provider/server/server.go

```diff
@@ -799,12 +794,25 @@ func (s *OCSProviderServer) ReportStatus(ctx context.Context, req *pb.ReportStat
 		return nil, status.Errorf(codes.Internal, "Failed to update lastHeartbeat payload in the storageConsumer resource: %v", err)
 	}
 
+	storageConsumer, err := s.consumerManager.Get(ctx, req.StorageConsumerUUID)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "Failed to get storageConsumer resource: %v", err)
+	}
+
 	channelName, err := s.getOCSSubscriptionChannel(ctx)
 	if err != nil {
 		return nil, status.Errorf(codes.Internal, "Failed to construct status response: %v", err)
 	}
+	var arr = make([]string, 0)
```

Comment:
1. It is better to use any when constructing the value to hash. So there will be no need to convert things to 
string
2. The structure needs to be stable, so don't use one item as a key and one item as a value. 
```suggestion 
	var arr = []any{}
```